odoo/addons/base/models/ir_attachment.py

```diff
@@ -343,10 +343,12 @@ def _check_contents(self, values):
         xml_like = 'ht' in mimetype or ( # hta, html, xhtml, etc.
                 'xml' in mimetype and    # other xml (svg, text/xml, etc)
                 not 'openxmlformats' in mimetype)  # exception for Office formats
+        xml_image_like = xml_like and 'image' in mimetype
         user = self.env.context.get('binary_field_real_user', self.env.user)
         if not isinstance(user, self.pool['res.users']):
             raise UserError(_("binary_field_real_user should be a res.users record."))
         force_text = xml_like and (
+            not (xml_image_like and self.env.context.get('attachments_mime_force_xml_image'))) and (
```

Comment:
@bso-odoo Didn't test or check, but seems not really safe as anyone (not only publisher) can pass a context quite easily, bypassing that security check, no?

Reply:
Now checking if the user "is somehow allowed to write xml images" - which is no by default, but is overridden in website to allow users with group `website.group_website_publisher`.